iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/client/IoTDBDataNodeAsyncClientManager.java

```diff
@@ -94,6 +98,7 @@ public IoTDBDataNodeAsyncClientManager(
       }
     }
     endPoint2Client = ASYNC_PIPE_DATA_TRANSFER_CLIENT_MANAGER_HOLDER.get(receiverAttributes);
+    attributesCount.compute(receiverAttributes, (k, v) -> v == null ? 1 : v + 1);
```

Comment:
Please use the same lock for put and remove operations, and ensure the atomicity of adding and increasing counts, removing and decreasing counts.

Reply:
@luoluoyuyu I tested it and found that using static + Concurrent can ensure thread safety.

Reply:
This step can only guarantee the atomicity of updating the reference. Here, it is necessary to ensure that the createClient and update reference steps are synchronized.